helm: remove topology RBAC section for cephfs deployment

humblec opened the pull request:
At present CephFS does not support topology aware provisioning and this
commit remove the same.

Updates: #3018

humblec committed:
helm: remove topology RBAC section for cephfs deployment

At present CephFS does not support topology aware provisioning and this
commit remove the same.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

Madhu-1 commented on charts/ceph-csi-cephfs/templates/provisioner-clusterrole.yaml:
```diff
@@ -53,12 +53,4 @@ rules:
     resources: ["persistentvolumeclaims/status"]
     verbs: ["update", "patch"]
 {{- end -}}
-{{- if .Values.topology.enabled }}
```
please remove the topology section in value.yaml also?

humblec replied:
@Madhu-1 sure , done

Madhu-1 commented on charts/ceph-csi-cephfs/values.yaml:
```diff
@@ -206,18 +206,6 @@ provisioner:
 # selinux-enabled filesystems
 selinuxMount: true
 
-topology:
-  # Specifies whether topology based provisioning support should
-  # be exposed by CSI
-  enabled: false
-  # domainLabels define which node labels to use as domains
-  # for CSI nodeplugins to advertise their domains
-  # NOTE: the value here serves as an example and needs to be
-  # updated with node labels that define domains of interest
-  domainLabels:
-    - failure-domain/region
-    - failure-domain/zone
```
Please update Readme also as these values are getting removed.

humblec replied:
@Madhu-1 where exactly you refer

Madhu-1 replied:
@humblec here https://github.com/ceph/ceph-csi/tree/devel/charts/ceph-csi-cephfs#configuration

humblec replied:
@Madhu-1 updated.thanks

humblec committed:
doc: remove topology mention from cephfs documentation

as we are removing the topology configuration from the deployment
this commit remove it from the documentation too.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

Madhu-1 reviewed (commented):
changes LGTM, @humblec do we need 2 commits for this one?

humblec commented:
> changes LGTM, @humblec do we need 2 commits for this one?

@Madhu-1  I am not sure either.. I am kind of lost on the arrangement of commits in this repo atleast :( . most of the time I was asked to keep it in different commits if the changes are at different code areas. so kept it in different .

Madhu-1 commented:
> > changes LGTM, @humblec do we need 2 commits for this one?
> 
> @Madhu-1 I am not sure either.. I am kind of lost on the arrangement of commits in this repo atleast :( . most of the time I was asked to keep it in different commits if the changes are at different code areas. so kept it in different .

Even am also, don't like different commits one for doc, one for the helm, and deployment builde.env,e2e  for the same change. this might cause issues in cherry-pick etc. lets discuss this in tomorrow's standup.

Madhu-1 reviewed (approved):
Approving this one for now as this pattern is followed and needs to be sorted out soon.

humblec commented:
@yati1998 ptal.. thanks.

nixpanic commented:
> Even am also, don't like different commits one for doc, one for the helm, and deployment builde.env,e2e for the same change. this might cause issues in cherry-pick etc. lets discuss this in tomorrow's standup.

My preference is to have commits address a single thing. In this case both commits are part of a small cleanup, which can definitely go together.

For large PRs, that introduce new features, helper function and their unit-tests can go together, consumption of the new functions (if complex enough) should probably go into its own commit, just like documentation with examples.

Splitting commits is done to help maintainers and contributors reviewing the PR and it updates, as well as checking the history of code while working on it at a later time.
